src/components/Filters/index.tsx

```diff
@@ -104,40 +105,55 @@ interface FiltersState {
 	searchString: string;
 	filters: JSONSchema6[];
 	views: FiltersView[];
+	schema: JSONSchema6;
 }
 
 class Filters extends React.Component<FiltersProps, FiltersState> {
 	constructor(props: FiltersProps) {
 		super(props);
-		const { filters, views } = this.props;
+		const { filters, schema, views } = this.props;
+
+		const flatSchema = SchemaSieve.flattenSchema(schema);
 
 		this.state = {
 			showModal: false,
 			searchString: '',
-			edit: [this.getCleanEditModel()],
 			editingFilter: null,
 			filters: filters || [],
 			views: views || [],
+			schema: flatSchema,
+			edit: [this.getCleanEditModel(null, flatSchema)],
 		};
+	}
 
-		// Clean exsting rules on load
+	componentWillReceiveProps(prevProps: FiltersProps) {
+		// If the schema prop updates, also update the internal 'flat' schema
+		if (!isEqual(prevProps.schema, this.props.schema)) {
+			this.setState({
+				schema: SchemaSieve.flattenSchema(this.props.schema),
+			});
+		}
 	}
 
-	getCleanEditModel(field?: string) {
-		const { schema } = this.props;
+	getCleanEditModel(field?: string | null, schema?: JSONSchema6) {
+		if (!schema) {
```

Comment:
Could this be in the method declaration args as a default value `schema: JSONSchema6 = this.state.schema`?